gobblin-cluster/src/test/java/org/apache/gobblin/cluster/suite/IntegrationJobCancelSuite.java

```diff
@@ -40,15 +42,23 @@
         ConfigurationKeys.SOURCE_CLASS_KEY, "org.apache.gobblin.cluster.SleepingCustomTaskSource",
         ConfigurationKeys.JOB_ID_KEY, JOB_ID,
         GobblinClusterConfigurationKeys.HELIX_JOB_TIMEOUT_ENABLED_KEY, Boolean.TRUE,
-        GobblinClusterConfigurationKeys.HELIX_JOB_TIMEOUT_SECONDS, 10L, SleepingTask.TASK_STATE_FILE_KEY, TASK_STATE_FILE))
-        .withFallback(rawJobConfig);
+        GobblinClusterConfigurationKeys.HELIX_JOB_TIMEOUT_SECONDS, 10L,
+
+        // Put SleepingTask in infinite sleep so that cancellation thereby ensuring cancellation to happen.
+        SleepingTask.TASK_STATE_FILE_KEY, TASK_STATE_FILE))
+        .withFallback(rawJobConfig.withValue(SLEEPING_TASK_SLEEP_TIME, ConfigValueFactory.fromAnyRef(-1)));
     return ImmutableMap.of(JOB_NAME, newConfig);
   }
 
+  /**
+   * Note This verification now ties to {@link SleepingTask} which is not ideal.
+   */
   @Override
   public void waitForAndVerifyOutputFiles() throws Exception {
+    // Sleeing task is in infinite sleeping, so unless an cancelling is hitting task execution, this line won't be printed.
```

Comment:
Change comment to: "SleepingTask is in an infinite sleep. The log line is printed only when a cancellation in invoked."